core/state-machine/src/testing.rs

```diff
@@ -42,27 +44,32 @@ pub struct TestExternalities<H: Hasher, N: ChangesTrieBlockNumber> {
 
 impl<H: Hasher, N: ChangesTrieBlockNumber> TestExternalities<H, N> {
 	/// Create a new instance of `TestExternalities`
-	pub fn new(inner: HashMap<Vec<u8>, Vec<u8>>) -> Self {
-		Self::new_with_code(&[], inner)
+	pub fn new(storage: StorageTuple) -> Self {
```

Comment:
I think we should revert this change and add a `new_with_child` function.

Reply:
hmm at the end I think user shouldn't differentiate top storage from child storages, also I feel like with children should be the default. I keep all the `From<top>` more for historical reason than because I want to. Also I can understand implementing your API

Reply:
If I don't want to initialize a child storage, I will give `Default::default()` anyway and it seems that in most cases we don't want to initialize the child storage to something meaningful.

Reply:
I made the change